### PR TITLE
JDK-8256208: Javadoc's generated overview does not show classes of unnamed package

### PR DESCRIPTION
**Pull request by hns**
This is a simple fix to include the unnamed package when javadoc generates documentation for both unnamed and named packages. As elsewhere in javadoc, the package is listed as "Unnamed Package" (subject to localization).

Also, when documenting only code in the unnamed package, the index.html page redirects to the overview page of the unnamed package instead of the first documented class, which is generally more adequate for an overview page.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8256208](https://bugs.openjdk.java.net/browse/JDK-8256208): Javadoc's generated overview does not show classes of unnamed package

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to 17f920cfcb43cb015876b2d274dc63cf4a699547

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5911/head:pull/5911` \
`$ git checkout pull/5911`

Update a local copy of the PR: \
`$ git checkout pull/5911` \
`$ git pull https://git.openjdk.java.net/jdk pull/5911/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5911`

View PR using the GUI difftool: \
`$ git pr show -t 5911`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5911.diff">https://git.openjdk.java.net/jdk/pull/5911.diff</a>

</details>

### CODE REVIEW

File: test/langtools/jdk/javadoc/doclet/testUnnamedPackage/src1/pkg/D.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package pkg;
+
+/**
+ * This is class D in the package pkg.
+ */
+public class D {}
```
Comment (jonathan-gibbons) [inline test/langtools/jdk/javadoc/doclet/testUnnamedPackage/src1/pkg/D.java:29]: add newline

File: test/langtools/jdk/javadoc/doclet/testUnnamedPackage/src1/pkg/package.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<html>
+    <body>
+        This is a package comment for package pkg.
+    </body>
+</html>
```
Comment (jonathan-gibbons) [inline test/langtools/jdk/javadoc/doclet/testUnnamedPackage/src1/pkg/package.html:5]: Old-style?   How quaint!

**Review by jonathan-gibbons — APPROVED** [review]
One new file needs a final newline.

File: test/langtools/jdk/javadoc/doclet/testUnnamedPackage/src1/pkg/package.html (hunk 2 above)
Comment (hns) [reply test/langtools/jdk/javadoc/doclet/testUnnamedPackage/src1/pkg/package.html:5]: One adapts to the environment... Also, not much missing for valid html5 except `<!doctype html>`

**Pull request closed without merging**